### PR TITLE
chore(mds): expand naive coverage across fields and widths

### PR DESCRIPTION
**Pull request by Fibonacci747**
`IntegratedCosetMds` and `CosetMds` tests only checked one field/size pair each, which left generic `F`/`const N` behavior under-tested and could miss edge cases.

Added parameterized naive-comparison tests for both MDS implementations across BabyBear and Goldilocks, and widths 1/2/4/8/16/32. Also replaced `from_u8(N as u8)` with `from_usize(N)` in test scaling.

### CODE REVIEW

File: mds/src/coset_mds.rs [hunk 1]
```diff
@@ -84,29 +84,57 @@ fn bowers_g_t<F: Field, A: Algebra<F>, const N: usize>(values: &mut [A; N], twid
...
-        const N: usize = 8;
-
+    fn matches_naive_for<F, const N: usize>()
+    where
+        F: TwoAdicField,
+        StandardUniform: Distribution<F>,
+    {
         let mut rng = SmallRng::seed_from_u64(1);
-        let mut arr: [F; N] = rng.random();
+        let mut arr: [F; N] = array::from_fn(|_| rng.random());
```
Comment (tcoratger) [inline mds/src/coset_mds.rs:106]:
```suggestion
        let mut arr: [F; N] = rng.random();
```

File: mds/src/integrated_coset_mds.rs [hunk 2]
```diff
@@ -68,34 +68,62 @@ impl<F: Field, A: Algebra<F>, const N: usize> MdsPermutation<A, N> for Integrate
...
-    #[test]
-    fn matches_naive() {
+    fn matches_naive_for<F, const N: usize>()
+    where
+        F: TwoAdicField,
+        StandardUniform: Distribution<F>,
+    {
         let mut rng = SmallRng::seed_from_u64(1);
-        let mut arr: [F; N] = rng.random();
+        let mut arr: [F; N] = array::from_fn(|_| rng.random());
```
Comment (tcoratger) [inline mds/src/integrated_coset_mds.rs:91]:
```suggestion
        let mut arr: [F; N] = rng.random();
```
